### DISCUSSION TITLE
Add native Windows support for Sail via PowerShell Scripts

### DISCUSSION THREAD

**Discussion by arcanefoam** [laravel/framework#48406] [Ideas] [answered]
As far as I understand, to use sail on Windows you need the Subsystem for Linux 2 (WSL2) because sail scripts are bash scripts. Native support would be possible if sail also provided Power Shell versions of the scripts.

I have translated the bash scripts to Power Shell and would like to know if this is a contribution you would be interested in accepting. If so, I would need some help to test if the scripts are working as expected (i.e. I dont know if you have a test suite that you run that I could run too). Additionally, I would need some guidance on where to actually put the scripts in the repo, as I think there is some file moving/copying to the bin folder during installation.

Thanks,

**Comment by vpratfr** [comment, marked as answer]
> because sail scripts are bash scripts

I think you require WSL2 as well because Docker requires it (and Sail requires Docker). I would be interested if there was a native Windows solution to run Docker. That would make it so much faster.

**Comment by arcanefoam** [comment]
I run docker on Windows without WSL2 and develop the PS scripts that way. 
![image](https://github.com/laravel/framework/assets/2157165/4e1511a3-d98b-4dfb-ace8-631e5e726d9b)
